pkg/clustersync/clusterSync.go

```diff
@@ -20,8 +39,228 @@ func ElectLeaderAndStart(ctx context.Context) {
 }
 
 func syncClusters(ctx context.Context) {
-	klog.Info("TODO: Start Sync clusters here.")
+	for {
+		select {
+		case <-ctx.Done():
+			klog.Info("Exit syncClusters.")
+			return
+		default:
+			klog.V(1).Info("Attempting to sync clusters.")
+			WatchClusters(ctx)
+		}
+	}
+}
+
+// Watches ManagedCluster objects and updates the database with a Cluster node.
+func WatchClusters(ctx context.Context) {
+	klog.Info("Begin ClusterWatch routine")
+
+	dynamicClient = config.GetDynamicClient()
+	dynamicFactory := dynamicinformer.NewDynamicSharedInformerFactory(dynamicClient,
+		time.Duration(config.Cfg.RediscoverRateMS))
+
+	// Create GVR for ManagedCluster and ManagedClusterInfo
+	managedClusterGvr, _ := schema.ParseResourceArg(managedClusterGVR)
+	managedClusterInfoGvr, _ := schema.ParseResourceArg(managedClusterInfoGVR)
+
+	//Create Informers for ManagedCluster and ManagedClusterInfo
+	managedClusterInformer := dynamicFactory.ForResource(*managedClusterGvr).Informer()
+	managedClusterInfoInformer := dynamicFactory.ForResource(*managedClusterInfoGvr).Informer()
+
+	// Create handlers for events
+	handlers := cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			klog.V(4).Info("clusterWatch: AddFunc")
+			processClusterUpsert(obj)
+		},
+		UpdateFunc: func(prev interface{}, next interface{}) {
+			klog.V(4).Info("clusterWatch: UpdateFunc")
+			processClusterUpsert(next)
+		},
+		// DeleteFunc: func(obj interface{}) {
```

Comment:
no delete?  Its ok if we handle all deletes with the next story, just checking.

Reply:
Yes, delete not implemented in this one